libs/designsystem/src/lib/components/text-link/text-link.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+      const icon = spectator.queryHost<HTMLElement>('kirby-icon');
+
+      const iconName = icon.getAttribute('name');
+      expect(iconName).toBe('link');
+    });
+
+    it('should render the icon to the right from the text', () => {
+      const icon = spectator.queryHost<HTMLElement>('kirby-icon');
+      const text = spectator.queryHost<HTMLElement>('a');
+      const textPosition = text.getBoundingClientRect();
+      const iconPosition = icon.getBoundingClientRect();
+      expect(textPosition.right).toBeLessThan(iconPosition.right);
+    });
+  });
+  describe('size and padding', () => {
+    interface TestCase {
+      size?: string;
+      expected: {
+        fontSize: string;
+        fontLineHeight: string;
+        iconPaddingLeft: string;
+        iconPaddingRight: string;
+        iconFontSize: string;
+        iconVerticalAlign: string;
+      };
+    }
+
+    const testCases: TestCase[] = [
+      {
+        size: 'md',
+        expected: {
+          fontSize: 'n',
+          fontLineHeight: 'n',
+          iconPaddingLeft: 'xxxs',
+          iconPaddingRight: 's',
+          iconFontSize: 'm',
+          iconVerticalAlign: 'middle',
+        },
+      },
+      {
+        size: 'sm',
+        expected: {
+          fontSize: 's',
+          fontLineHeight: 's',
+          iconPaddingLeft: 'xxxs',
+          iconPaddingRight: 's',
+          iconFontSize: 'm',
+          iconVerticalAlign: 'middle',
+        },
+      },
+      {
+        size: 'xs',
+        expected: {
+          fontSize: 'xs',
+          fontLineHeight: 'xs',
+          iconPaddingLeft: 'xxxxs',
+          iconPaddingRight: 's',
+          iconFontSize: 's',
+          iconVerticalAlign: 'middle',
+        },
+      },
+      {
+        size: '',
+        expected: {
+          fontSize: 'n',
+          fontLineHeight: 'n',
+          iconPaddingLeft: 'xxxs',
+          iconPaddingRight: 's',
+          iconFontSize: 'm',
+          iconVerticalAlign: 'middle',
+        },
+      },
+    ];
+    testCases.forEach((scenario) => {
+      describe(`when size is ${scenario.size || 'empty'}`, () => {
+        describe(`external link`, () => {
+          beforeEach(() => {
+            spectator = createHost(
+              `<kirby-text-link size='${scenario.size}' text='External link' link='https://angular.io/api/router/RouterLink'></kirby-text-link>`
+            );
+          });
+          it('should render icon with correct styling', () => {
+            const icon = spectator.queryHost<HTMLElement>('kirby-icon');
+
+            expect(icon).toHaveComputedStyle({
+              'font-size': size(scenario.expected.iconFontSize),
+              'padding-left': size(scenario.expected.iconPaddingLeft),
+              'padding-right': size(scenario.expected.iconPaddingRight),
+              'vertical-align': scenario.expected.iconVerticalAlign,
+            });
+          });
+          it('should render anchor link with corrrect styling', () => {
```

Comment:
Consider readability:
```suggestion
          });
          
          it('should render anchor link with corrrect styling', () => {
```